PRODDOCS-816: Snapshots (#9118)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/snapshots.md

8.  Snapshots are only useful if you run them frequently &mdash; schedule the `dbt snapshot` command to run regularly.

>     Snapshots capture the state of your data at the time they run. If a record changes multiple times between snapshot runs, snapshots may not record intermediate changes. Choose a schedule that matches the change rate you need to analyze.

Reviewer: Hey @nataliefiann I feel like this should possibly become a reusable and the info can be communicated consistently in the other places you've updated below. Wha do you think?

>     Snapshots capture the state of your data at the time they run. If a record changes multiple times between snapshot runs, snapshots may not record intermediate changes. Choose a schedule that matches the change rate you need to analyze.

Reviewer: I tried to rephrase this by what it does capture. But I'm not sure if this is correct?
```suggestion
    A snapshot captures the state of your data when it's run. Even when a record changes multiple times between snapshot runs, the snapshot only captures the most recent state and none of the intermediate changes. Choose a schedule that matches the change rate you need to analyze.
```

Author: Hiya @runleonarun 

Thanks for the suggestion on rephrasing. I tried to keep the wording aligned with how snapshots are described in our docs: snapshots build a table from the snapshot query result set and then, on subsequent runs, compare what’s returned at each run to record new versions ([How snapshots work](https://docs.getdbt.com/docs/build/snapshots?version=2.0#how-snapshots-work) section). For the timestamp strategy specifically, we determine change by whether `updated_at` is more recent than the last snapshot run. Because of that, I think the most precise phrasing is “state observable at run time” rather than “most recent state,” since dbt can’t see or guarantee capture of every intermediate transition that happened between runs.
